mantle/cmd/kola/options.go

```diff
@@ -63,6 +63,7 @@ func init() {
 	bv(&kola.NoNet, "no-net", false, "Don't run tests that require an Internet connection")
 	ssv(&kola.Tags, "tag", []string{}, "Test tag to run. Can be specified multiple times.")
 	bv(&kola.Options.SSHOnTestFailure, "ssh-on-test-failure", false, "SSH into a machine when tests fail")
+	bv(&kola.Options.SuppressDefaultChecks, "no-default-checks", false, "Disable default checks for failed systemd units and SELinux AVC denials")
```

Comment:
Thinking more on this; WDYT about splitting this into two separate switches? One for systemd units and one for SELinux denials.

Reply:
To elaborate: I think eventually we should have precise control over what checks we want to allow to fail. Refining that can come later (because it requires more kola work), but at least right now we already have two broad categories we can split on.

Reply:
Hmm.  Agree that unit failures are usually much worse.  But making it more granular feels tricky because we're likely to add more here.  A good example is systemd generator failures and systemd ordering failures are also missed today.
And another good one is coredumps (which usually result in a unit failure, but not always).

Maybe what we really want is a syntax to ignore specific AVC denials or unit failures?  Something like:


```
# This file documents currently known-to-fail kola tests. It is consumed by
# coreos-assembler to automatically skip some tests. For more information,
# see: https://github.com/coreos/coreos-assembler/pull/866.
tests:
  - pattern: fcos.internet
    tracker: https://github.com/coreos/coreos-assembler/pull/1478
  - pattern: podman.workflow
    tracker: https://github.com/coreos/coreos-assembler/pull/1478
avc:
  # 3 tuple is string match on (vector, scontext, tcontext)
  - read chronyd_t systemd_resolved_var_run_t
units:
  # Failing on vsphere, but not a blocker
  - NetworkManager.service
```

Reply:
> But making it more granular feels tricky because we're likely to add more here.

True. But while having a single switch is useful for dev, it's too blunt an instrument for CI/prod because it makes it too easy to miss out on regressions because of e.g. one AVC we're working around. (This was part of the motivation for https://github.com/coreos/coreos-assembler/pull/2064.)

> Maybe what we really want is a syntax to ignore specific AVC denials or unit failures? Something like:

Yes, 100%. (This is what I mean with https://github.com/coreos/coreos-assembler/pull/1920#issuecomment-782337257; though yeah it'd be cleaner to first make the top-level object a dict rather than a list).

Reply:
And probably before that, move this denylist directly into kola.

Reply:
If we're willing to blanket-ignore denials/failures in specific tests, we have the existing test flags mechanism for this.  It doesn't currently support more granular filters, but in principle it could.